src/views/DatasetManager/components/DatasetOverview/components/CategoryAccordion/components/CategoryDetailsDialog/CategoryDetailsDialog.js

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useCallback, useEffect, useMemo, useState, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { Grid, Typography, Dialog, DialogTitle, Box, Button, DialogContent } from '@mui/material';
+import { TABLE_DATA_STATUS } from '@cosmotech/ui';
+import { GenericTable, ApplicationErrorBanner } from '../../../../../../../../components/';
+import { TranslationUtils } from '../../../../../../../../utils';
+import GraphIndicator from '../../../GraphIndicator';
+import { useCategoryDetailsDialogHook } from './CategoryDetailsDialogHook';
+
+export const CategoryDetailsDialog = (props) => {
+  const { kpis, category } = props;
+
+  const {
+    datasetName,
+    datasetId,
+    getQuery,
+    isDarkTheme,
+    applicationError,
+    setApplicationErrorMessage,
+    clearApplicationErrorMessage,
+  } = useCategoryDetailsDialogHook();
+  const { t } = useTranslation();
+
+  const [open, setOpen] = useState(false);
+  const previousAppError = useRef(applicationError);
+  useEffect(() => {
+    if (open) {
+      previousAppError.current = applicationError;
+      clearApplicationErrorMessage();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open, clearApplicationErrorMessage]); // Do not include applicationError in deps list
+
+  const handleOpen = useCallback(() => {
+    setOpen(true);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    if (previousAppError.current != null) setApplicationErrorMessage(previousAppError.current);
+    else clearApplicationErrorMessage();
```

Comment:
Just a question: now, when all errors are handled in the placeholder, do we still need this logic with previous errors?